Criterion/Main/Options.hs

```diff
@@ -154,12 +159,14 @@ match :: ReadM MatchType
 match = do
   m <- readerAsk
   case map toLower m of
-    mm | mm `isPrefixOf` "pfx"    -> return Prefix
-       | mm `isPrefixOf` "prefix" -> return Prefix
-       | mm `isPrefixOf` "glob"   -> return Glob
-       | otherwise                ->
-         readerError $ show m ++ " is not a known match type. "
-                              ++ "Try \"prefix\" or \"glob\"."
+    mm | mm `isPrefixOf` "pfx"      -> return Prefix
+       | mm `isPrefixOf` "prefix"   -> return Prefix
+       | mm `isPrefixOf` "glob"     -> return Glob
+       | mm `isPrefixOf` "pattern"  -> return Pattern
```

Comment:
I decided to keep the current behavior of preferring `prefix` over `pattern`, since I don't know how many people are going to rely on this parsing quirk in the first place. If someone complains, we can change this.